--- old/ParserElements.inc	2007-04-24 17:30:50.000000000 -0400
+++ new/ParserElements.inc	2007-11-17 00:34:47.000000000 -0500
@@ -1089,7 +1089,11 @@
             $oldx = $x;
             $x = &$par;
         }
-        if (is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) {
+        if ((is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) ||
+            (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass'))) {
+            if (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass')) {
+                $oldx = $x;
+            }
             $extends = $oldx->getExtends(true);
             foreach ($extends->getMethods() as $method) {
                 $var = new parserMethod($oldx->getExtends());
@@ -1187,7 +1191,11 @@
             $oldx = $x;
             $x = &$par;
         }
-        if (is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) {
+        if ((is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) ||
+            (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass'))) {
+            if (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass')) {
+                $oldx = $x;
+            }
             $extends = $oldx->getExtends(true);
             foreach ($extends->getProperties() as $property) {
                 $var = new parserVar($oldx->getExtends());
@@ -1263,7 +1271,11 @@
             $oldx = $x;
             $x = &$par;
         }
-        if (is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) {
+        if ((is_a($oldx, 'parserClass') && is_a($oldx->getExtends(true), 'ReflectionClass')) ||
+            (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass'))) {
+            if (is_a($x, 'parserClass') && is_a($x->getExtends(true), 'ReflectionClass')) {
+                $oldx = $x;
+            }
             $extends = $oldx->getExtends(true);
             if (!$extends->getConstants()) {
                 return $arr;
